### ISSUE TITLE
Mode "edge" vs "edges" in gammapy.maps

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1738]
We currently have `mode="edges"` here:
```
$ ack \"edges\" gammapy/maps
gammapy/maps/wcs.py
558:        if mode == "edges":
604:        if mode == "edges":
770:        coord = self.get_coord(mode="edges")
```
and `mode="edge"` here:
```
$ ack \"edge\" gammapy/maps
gammapy/maps/geom.py
364:    def __init__(self, nodes, interp="lin", name="", node_type="edge", unit=""):
372:        if node_type == "edge":
460:        node_type = kwargs.setdefault("node_type", "edge")
462:        if node_type == "edge":
522:        return cls(edges, node_type="edge", **kwargs)
605:        if self._node_type == "edge":
625:        vals = self.edges if self.node_type == "edge" else self.center
1359:            if ax.name == "energy" and ax.node_type == "edge":
1363:            elif ax.node_type == "edge":

gammapy/maps/hpxnd.py
232:        elif mode in ["edge", "interp"]:
235:            vals = self.interp_by_coord(coords, interp=0 if mode == "edge" else order)

gammapy/maps/wcsnd.py
338:        elif mode in ["edge", "interp"]:
342:            vals = self.interp_by_coord(coords, interp=0 if mode == "edge" else order)

gammapy/maps/base.py
962:            if axis.node_type == "edge":
```

I'm not sure if that got all places or if the search is somehow incomplete.

It looks like actually there's a consistency to this at the moment: "edge" is used for MapAxis and "edges" for spatial coordinates?

Still, this might be confusing for users and we could change to uniform "edge" everywhere, given that the "edge" mode in the coords is a recent addition.

@adonath - Thoughts?

**Comment by adonath** [comment]
@cdeil I agree we should go for consistency here.  Whether we choose `edge` or `edges` I don't really care (have a slight preference for `edges`, as there is also the `MapAxis.edges` attribute...). Have you checked whether the mode `center` is actually consistent? Or are there any `centers` used in Gammapy?

I just noticed that for the node edges we have `MapAxis.edges` while for the node centers we've chosen the singular term `MapAxis.center`. I'd also like to introduce consistency here. For the attributes I have a clear preference for the plural term, because it is actually an array with multiple values returned.

**Comment by cdeil** [comment]
I remember I had a discussion with @woodmd at the start of gammapy.maps whether to use plural `coords`, `pixs` and `idxs` or singular `coord`, `pix` and idx`. We decided to go with singular, even if those are arrays. That's not uncommon, e.g. Time and SkyCoord can represent arrays. Most of gammapy.maps is consistent with `coord`, `pix` and `idx`; I think all of the method names are, and for the argument and local variable names it's still a mix, but I think this could be cleaned up to all-singular.

So for consistency with that, and because I do prefer `idx` and `pix` over `idxs` and `pixs`, I'd also suggest to go for singular `edge`.

Looks like this won't break fermipy; they do have edges, but only on their castro class:
https://github.com/fermiPy/fermipy/search?p=2&q=edges&unscoped_q=edges

I'll come over now so that we can discuss and decide offline.
